cmd/assaf-http/main.go

```diff
@@ -310,53 +263,54 @@ func subscribe(queueName string, sessions chan chan session, consumeRequests cha
 
 		log.Printf("subscribed...")
 
-		curr := 0
-		seen := 0
-		for msg := range deliveries {
-			for consumeRequest := range consumeRequests {
-
-				log.Printf("got msg in deliveries! %v", msg.Body)
-				log.Println(strconv.Atoi(msg.RoutingKey))
-				routingKey, _ := strconv.Atoi(msg.RoutingKey)
-				if curr == 0 {
-					curr = consumeRequest.reqId
-				} else if curr == consumeRequest.reqId {
-					seen = seen + 1
-				}
-
-				if consumeRequest.reqId == routingKey {
-					spCtx, _ := amqptracer.Extract(msg.Headers)
-					sp := opentracing.StartSpan(
-						"ConsumeMessage",
-						opentracing.FollowsFrom(spCtx),
-					)
-					defer sp.Finish()
-
-					// Update the context with the span for the subsequent reference.
-					ctx = opentracing.ContextWithSpan(ctx, sp)
-
-					consumeRequest.reqChannel <- message{msg.Body, msg.RoutingKey}
-					sub.Ack(msg.DeliveryTag, false)
-					break
-				}
-				// Return message to queue if it's not stale yet
-				if isMsgFresh(consumeRequest) {
-					consumeRequests <- consumeRequest
-				} else {
-					sub.Ack(msg.DeliveryTag, false)
-				}
-
-				// break loop after going through all consume requests once
-				if seen > 0 {
```

Comment:
this is an especially ugly hack, sometimes the msgs and reqs would go out of sync and the inner loop will go to an infinite loop since none of the conditions were met so I was counting number of times we encountered it. this is obviously not a scalable solution